### PR TITLE
Add support for Qtum

### PR DESCRIPTION
**Pull request by timemarkovqtum**
## Description

Added support for send/receive `Qtum` coins.
Support exist for `P2PKH` and `bech32`.
Support for `P2SH` added but disabled due to `Litecoin` and `Qtum` have the same prefix for it, more description in the commits.
Support for smart contracts not added, due to no such functionalities in the trust wallet.
Tested only with unit tests.

## Testing instructions

Run `./bootstrap.sh` to execute the unit tests for `Qtum`.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: include/TrustWalletCore/TWP2PKHPrefix.h [hunk 1]
```diff
@@ -21,6 +21,7 @@ enum TWP2PKHPrefix {
     TWP2PKHPrefixGroestlcoin = 0x24,
     TWP2PKHPrefixZcoin = 0x52,
     TWP2PKHPrefixZcashT = 0xB8,
+    TWP2PKHPrefixQtum = 0x3a,
```
Comment (alejandro-isaza) [inline include/TrustWalletCore/TWP2PKHPrefix.h]: Please sort the cases

File: src/Coin.cpp [hunk 2]
```diff
@@ -144,6 +148,7 @@ TWPurpose TW::purpose(TWCoinType coin) {
     case TWCoinTypeNEO:
     case TWCoinTypeKIN:
     case TWCoinTypeTheta:
+    case TWCoinTypeQtum:
```
Comment (alejandro-isaza) [inline src/Coin.cpp]: We should use BIP44

Comment (vikmeup) [reply src/Coin.cpp]: I think it's already using BIP44?

Comment (alejandro-isaza) [reply src/Coin.cpp]: Yes, my mistake :)

**Comment by alejandro-isaza** [comment]
Can you please resolve the conflicts?

**Comment by timemarkovqtum** [comment]
Sort prefix and rebased

**Comment by hewigovens** [comment]
@timemarkovqtum could you please confirm the derivation path is compatible with other mobile wallets? sorry for another rebase request, we just re-enabled iOS CI (for Swift 5)

**Comment by timemarkovqtum** [comment]
@hewigovens  According to the source code check both Android and iOS Qtum wallets are using the derivation path used in Qtum Desktop Wallet `m/88'/0'/`
https://github.com/qtumproject/qtum-ios/blob/develop/qtum%20wallet/Spendable%20Controll%20Flow/Wallet/Wallet.m#L265
https://github.com/qtumproject/qtum-android/blob/develop/app/src/main/java/org/qtum/wallet/datastorage/KeyStorage.java#L108-L109

**Comment by vikmeup** [comment]
@timemarkovqtum what kind of standard are you guys following? It does not look like BIP44 standard to me.

**Comment by timemarkovqtum** [comment]
The standards are inherited from Bitcoin:
https://github.com/bitcoin/bitcoin/blob/master/src/wallet/wallet.cpp#L271
There is just change from 0 to 88 in the first index.

**Comment by hewigovens** [comment]
why not use bip44 and slip44? 88 doesn't make any sense

https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
https://github.com/satoshilabs/slips/blob/master/slip-0044.md

and looks like qtum supports segwit, how do you guys deal with bip84?

**Comment by timemarkovqtum** [comment]
According to bip checks, the derivation scheme correspond to:
https://github.com/bitcoin/bips/blob/master/bip-0032/derivation.png
Bitcoin desktop wallet looks like is using that derivation, probably due to backward compatibility, that result in using other desktop wallets too.
BIP44, BIP49, and BIP84 are all based on BIP32.

**Pull request merged**